internal/appstore/appstore.go

```diff
@@ -0,0 +1,83 @@
+package appstore
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/saucelabs/saucectl/internal/storager"
+)
+
+// AppStore implements functions for AppStore interface
+type AppStore struct {
+	HTTPClient *http.Client
+	URL        string
+	Username   string
+	AccessKey  string
+}
+
+// New returns an implementation for AppStore
+func New(url, username, accessKey string) storager.Storager {
+	return &AppStore{
+		HTTPClient: &http.Client{},
+		URL:        url,
+		Username:   username,
+		AccessKey:  accessKey,
+	}
+}
+
+// Upload uploads file to remote storage
+func (s *AppStore) Upload(fileName, formType string) error {
+	body, writer, err := prepareFile(fileName, formType)
+	if err != nil {
+		return err
+	}
+	defer writer.Close()
+
+	request, err := createRequest(s.URL, s.Username, s.AccessKey, body, writer)
+	if err != nil {
+		return err
+	}
+
+	resp, err := s.HTTPClient.Do(request)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	_, err = ioutil.ReadAll(resp.Body)
+	return err
+}
+
+func prepareFile(fileName, formType string) (*bytes.Buffer, *multipart.Writer, error) {
+	file, err := os.Open(fileName)
+	if err != nil {
+		return nil, nil, err
+	}
+	defer file.Close()
+
+	body := &bytes.Buffer{}
+	writer := multipart.NewWriter(body)
+	part, err := writer.CreateFormFile(formType, filepath.Base(file.Name()))
+	if err != nil {
+		return nil, nil, err
+	}
+	io.Copy(part, file)
+
+	return body, writer, nil
```

Comment:
Are you sure you want to return the writer here? `createRequest` is really just interested in the content type further down the line.

Reply:
Oh!! I understand that!